src/util/coordinate_calculation.cpp

```diff
@@ -27,26 +27,34 @@ class CheapRulerContainer
   public:
     CheapRulerContainer(const int number_of_rulers)
         : cheap_ruler_cache(number_of_rulers, mapbox::cheap_ruler::CheapRuler(0)),
-          step(90.0 / number_of_rulers)
+          step(90.0 * COORDINATE_PRECISION / number_of_rulers)
     {
         for (int n = 0; n < number_of_rulers; n++)
         {
             cheap_ruler_cache[n] = mapbox::cheap_ruler::CheapRuler(
-                step * (n + 0.5), mapbox::cheap_ruler::CheapRuler::Meters);
+                step * (n + 0.5) / COORDINATE_PRECISION, mapbox::cheap_ruler::CheapRuler::Meters);
         }
     };
-    mapbox::cheap_ruler::CheapRuler &getRuler(const double lat)
+
+    mapbox::cheap_ruler::CheapRuler &getRuler(const FixedLatitude lat_1, const FixedLatitude lat_2)
     {
-        return cheap_ruler_cache[std::min((int)std::floor(abs(lat) / step),
-                                          (int)cheap_ruler_cache.size() - 1)];
+        auto lat = (lat_1 + lat_2) / util::FixedLatitude{2};
+        return getRuler(lat);
+    }
+
+    mapbox::cheap_ruler::CheapRuler &getRuler(const FixedLatitude lat)
+    {
+        BOOST_ASSERT(step > 1);
+        auto bin = (std::abs(static_cast<int>(lat)) - 1) / step;
```

Comment:
What if the request is for `lat = 0` here?  This'll lead to `bin = -1`.

Reply:
Good catch! Didn't realize rounding direction was not actually defined for `1/2`.